crates/sui-types/src/base_types.rs

```diff
@@ -309,6 +309,17 @@ impl MoveObjectType {
         }
     }
 
+    pub fn try_extract_field_value(&self) -> SuiResult<TypeTag> {
```

Comment:
I kind of hate that this very specific function exists on this very generic type, but I can't deny that that seems to be the pattern here...  We can add that to the pile of things to clean up later -- it seems like it's only used for indexing purposes.